Competition2017/src/telemetry/InfluxDBConnection.java

```diff
@@ -0,0 +1,48 @@
+package telemetry;
+
+import java.util.concurrent.TimeUnit;
+
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.Point;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import xbot.common.properties.BooleanProperty;
+import xbot.common.properties.StringProperty;
+import xbot.common.properties.XPropertyManager;
+
+@Singleton
+public class InfluxDBConnection {
+
+    protected final BooleanProperty isEnabledProp;
+    protected final StringProperty influxAddressProp;
+    
+    protected final InfluxDB influxDB;
+    protected final String dbName = "XbotRobotData";
+    
+    @Inject
+    public InfluxDBConnection(XPropertyManager propMan) {
+        isEnabledProp = propMan.createPersistentProperty("InfluxDB logging enabled", false);
+        influxAddressProp = propMan.createPersistentProperty("Influx address", "http://10.4.88.100:8086");
```

Comment:
Two years ago, I implemented a web server that ran from a robot program process and served a web interface which let you control motors remotely. It had no dependencies on additional JARs (although I did later re-implement some of the communication logic with a more efficient library), required no additional set-up and did not impact load times _at all_. My benchmarking revealed that both CPU and memory were immeasurably affected by this system when it was in use, and when no client was connected there was absolutely no impact on the operation of the rest of the processor – a simple in-memory Boolean comparison told you if a client was there to consume data. No external requests were being made and no resources were being consumed. This implementation was a project that I worked on for a couple days in my free time at school, where almost all of that development effort was spent implementing fancy client-side features. We don’t need to speculate over what technical impact the PRed implementation has or what advantages it holds over other solutions, and we have no reason to attempt to convince ourselves that this is the optimal solution: we know empirically and demonstrably that a system like this can be implemented with ease with a fantastically small footprint and no downsides. So, instead of chasing the “coolest” implementation, I think we should be working towards an efficient, simple and robust solution. If we can't find an existing one that fits our needs right now, we shouldn't push to compromise – it’s a replacement for a system that we and thousands of other teams have used for years, and it continues to function well to this day. After all, implementing a browser-based solution for this very problem was something I advocated for in our summer planning and it was shot down within the hour; it can't be that important.